features/nanostack/nanostack-hal-mbed-cmsis-rtos/ns_event_loop_mutex.h

```diff
@@ -0,0 +1,14 @@
+/*
+ * Copyright (c) 2016 ARM Limited, All Rights Reserved
+ */
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+void ns_event_loop_mutex_init(void);
```

Comment:
Should this have doxygen doc (I noticed a comment in the code file, can be moved here)

Reply:
Don't think so - it's internal API. Only exists because of a split between two C files here.

Ultimately gets called a couple of layers down inside public APIs that do have Doxygen (`ns_hal_init` for Nanostack HAL-using apps like mbed client, else the constructor of a Nanostack-based `NetworkInterface`).

Reply:
We could use /internal and still move that comment here. Not certain if this is our practice (I searched platform/drivers for internal keyword, not found it in the docs) but would encourage that.

@ARMmbed/mbed-docs documentation for internal functions? we have doxygen internal flag set to NO so will be excluded. No mention in https://docs.mbed.com/docs/mbed-os-api-reference/en/latest/APIs/API_Documentation/